### ISSUE TITLE
Reconsider whether ARIA should ship new features that have incomplete API mappings and missing AT support.

### ISSUE THREAD

**Issue by cookiecrook** [w3c/aria#2361]
Reconsider whether ARIA should ship new features that have incomplete API mappings and missing AT support.

This problem came to my attention due to some Interop 2024 failures in [the aria-attribute-reflection test](https://wpt.fyi/results/html/dom/aria-attribute-reflection.html?label=master&label=experimental&aligned&view=interop&q=label%3Ainterop-2024-accessibility) for aria-colindextext and aria-rowindextext, which are new attributes in ARIA 1.3

WebKit engineers chose not to implement these two because there is no API mapping for it on Mac, and no near-term expectation for AT support. As such, the other browsers on Mac are reflecting the DOM properties, but do nothing with them other than pass the WPT test. I have an [open WPT PR to remove those tests from Interop](https://github.com/web-platform-tests/interop-accessibility/issues/151), since they were added mid-year.

It's also not clear from the Core-AAM listings whether there is AT support, or whether the Windows API mappings are speculative for Windows screen readers. (If so, I have concern with that approach overall.) I'll file a separate issue for Core-AAM to do a better job of indicating whether there is known downstream AT support. That's a larger feature request for the spec that we should discuss as a group.

As a related datapoint, Apple has proposed several ARIA features that have support in Apple AT, that the group has been reluctant to ship since Windows ATs are missing related features. The following are a few examples of those ARIA proposals that have not shipped because no Windows AT has similar feature support yet:

- #1215
- #744

The group should discuss whether the same should apply to these.... Especially if it turns out there is no Windows or Linux AT support and these are just speculative API mappings.

**Comment by spectranaut** [comment]
@joanmarie does Orca support these attributes, aria-colindextext and aria-rowindextext?

**Comment by joanmarie** [comment]
> @joanmarie does Orca support these attributes, aria-colindextext and aria-rowindextext?

Yup, it has for quite some time. It also does a fallback check for `aria-coltext` and `aria-rowtext` because (last time I checked) Google Sheets still exposes that instead of the (imho) more-clearly-named versions in the spec.

As a related aside, LibreOffice Calc followed the ARIA lead and exposes the row and column labels/indices via IA2 and ATK using the `colindextext` and `rowindextext` object attributes (this is how the ARIA properties are expected to be exposed when used in web content).

**Comment by stes-acc** [comment]
The practical thing about rowindextext is that you can use it to declare lead object names for a row.

This is of great use if users do vertical navigation in grids where rows represent object attributes. Instead of giving just the row number it can be a specific object name (of a product, of a person etc.). Addiotionally there is no need to define a specific row header for that row to achieve the same effect. It adds just ease of orientation without having read the entire row content at will (special Jaws/NVDA keys for that are available but they will always read more: parts of row or entire row depending on focus position in row).

**Comment by spectranaut** [comment]
This was discussed briefly in triage: https://www.w3.org/2024/10/24-aria-minutes.html#7fe9

@aleventhal mentioned there is support in NVDA and JAWS.

**Comment by cookiecrook** [comment]
Okay. I'll see if we can get mappings agreed upon for 1.3 even the implementation work doesn't happen in time for 1.3... Probably a new issue. 

However, this issue should be still be discussed among the editors... I don't think a Rec-track doc should get close to CR with "TBD" listed anywhere.

**Comment by cookiecrook** [comment]
FYI Core-AAM #2362 is ready for merge... 
And the Interop tests that were added mid-year have been marked as tentative until 2025.

**Comment by spectranaut** [comment]
This issue was discussed at length at today's meeting: https://www.w3.org/2024/10/31-aria-minutes.html#0352

We got closer to considering having editors notes explain the status of implementations. There was still some concern about this. @jnurthen volunteered to make an issue about the notes, however, we should discuss further before implementing.
